algo-core/scheduler_utils.py

```diff
@@ -0,0 +1,156 @@
+import argparse
+import json
+import sys
+from pathlib import Path
+import jsonschema
+import pandas as pd
+
+
+# Input filenames:
+filename_onboarding = "onboarding_agents.txt"
+filename_mentors = "mentors.txt"
+filename_new = "new_agents.txt"
+input_folder = ""
+
+def get_project_root() -> str:
+    return str(Path(__file__).parent.parent)
```

Comment:
```suggestion
def get_project_root() -> Path:
    return Path(__file__).parent.parent
```

Any chance we could use the `Path` objects directly? They're built specifically for handling path-related tasks, and file IO operations normally accept them just as well as strings.
It looks like currently we're converting between `Path` and `str` repeatedly for no good reason. Am I missing anything?

Reply:
I had an issue with that I remember, that's why I specifically changed the return type. let me look into it further what the issue was and probably you can offer a better solution

Reply:
Cool. All reputable libraries handling file IO should accept `Path` equally well since it's been around for a while now. But let me know if anything weird comes up.